app/src/main/java/com/poupa/vinylmusicplayer/model/Album.java

```diff
@@ -28,25 +30,29 @@ public long getId() {
     }
 
     public String getTitle() {
-        return safeGetFirstSong().albumName;
+        String name = safeGetFirstSong().albumName;
+        if (MusicUtil.isAlbumNameUnknown(name)) {
+            return UNKNOWN_ALBUM_DISPLAY_NAME;
+        }
+        return name;
     }
 
     public long getArtistId() {
         return safeGetFirstSong().artistId;
     }
 
     public String getArtistName() {
-        return safeGetFirstSong().artistName;
+        String name = safeGetFirstSong().artistName;
+        if (MusicUtil.isArtistNameUnknown(name)) {
+            return Artist.UNKNOWN_ARTIST_DISPLAY_NAME;
```

Comment:
why not keep this at the Discography level, before it is inserted to the DB?

Reply:
Hmm, that's possible. This implies having rawTitle/rawArtistName/rawAlbumName vs displayTitle/displayArtistName/displayAlbumName fields. Memory overhead vs CPU overhead.

We still need the rawXXX things for internal sorting/comparison/hash code generation I think.